lib/alternatives.js

```diff
@@ -81,7 +81,7 @@ internals.Alternatives.prototype.when = function (ref, options) {
     var obj = this.clone();
     var is = Cast.schema(options.is);
 
-    if (options.is === null || !options.is.isJoi) {
```

Comment:
I did it this way to not break the unit test on the ``null`` case: https://github.com/ludoblues/joi/blob/master/test/alternatives.js#L216

The expected behavior between ``null`` and ``undefined`` is not the same according to this test.
Should the test on ``null`` be changed to work the same way ?